tests/integrations/object_detection/test_object_detection.py

```diff
@@ -0,0 +1,242 @@
+        # Test that file is created
+        manager = integration_manager
+        model_file = os.path.join(manager.save_dir.name, "exp", "weights", "last.pt")
+        assert os.path.isfile(model_file)
+
+        # Test that model file loadable
+        _, extras = load_checkpoint(
+            type_="val", weights=model_file, device=torch.device("cpu")
+        )
+
+        # Test that training ran to completion
+        assert extras["ckpt"]["epoch"] == -1
+
+    @skip_inactive_stage
+    def test_train_metrics(self, integration_manager):
+        # Test train metric(s) if specified
+        manager = integration_manager
+        train_args = manager.configs["train"]
+        results_file = os.path.join(manager.save_dir.name, "exp", "results.csv")
+
+        if "target_name" in train_args.test_args:
+            train_test_args = train_args.test_args
+            results = pd.read_csv(results_file, skipinitialspace=True)
+
+            metric_key = METRIC_TO_COLUMN[train_test_args["target_name"]]
+            metric = results[metric_key].iloc[-1] * 100
+
+            target_mean = train_test_args["target_mean"]
+            target_std = train_test_args["target_std"]
+
+            assert target_mean - target_std <= metric <= target_mean + target_std
+
+    @skip_inactive_stage
+    def test_export_onnx_graph(self, integration_manager):
+        # Test that onnx model is loadable and passes onnx checker
+        manager = integration_manager
+        export_args = manager.configs["export"]
+        onnx_file = os.path.join(
+            os.path.dirname(export_args.run_args.weights), "last.onnx"
+        )
+
+        assert os.path.isfile(onnx_file)
+
+        model = onnx.load(onnx_file)
+        onnx.checker.check_model(model)
+
+    @skip_inactive_stage
+    def test_export_target_model(self, integration_manager):
+        # If target model provided in target_args, test that they have similar
+        # ort output
+        manager = integration_manager
+        export_args = manager.configs["export"]
+        target_model_path = export_args.test_args.get("target_model")
+
+        if not target_model_path:
+            pytest.skip("No target model provided")
+
+        export_model_path = os.path.join(
+            os.path.dirname(export_args.run_args.weights), "last.onnx"
+        )
+
+        # Downloads model if zoo stubs and additionally tests that it can be loaded
+        _, *_ = load_checkpoint(
+            type_="val", weights=target_model_path, device=torch.device("cpu")
+        )
+
+        _test_model_op_counts(export_model_path, target_model_path)
+
+        compare_outputs = export_args.test_args.get("compare_outputs", True)
+        if isinstance(compare_outputs, str) and (
+            compare_outputs.lower() in ["none", "False"]
+        ):
+            compare_outputs = False
+        if compare_outputs:
+            _test_model_inputs_outputs(export_model_path, target_model_path)
+
+
+_TEST_OPS = {
```

Comment:
noting we should definitely have this as a shared function in a follow up

Reply:
Yep, it's on my list of follow up updates